operator/pkg/ingress/service.go

```diff
@@ -0,0 +1,221 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright Authors of Cilium
+
+package ingress
+
+import (
+	"fmt"
+
+	"github.com/sirupsen/logrus"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/intstr"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/client-go/tools/cache"
+	"k8s.io/client-go/util/workqueue"
+
+	"github.com/cilium/cilium/pkg/k8s"
+	"github.com/cilium/cilium/pkg/k8s/informer"
+	slim_corev1 "github.com/cilium/cilium/pkg/k8s/slim/k8s/api/core/v1"
+	slim_networkingv1 "github.com/cilium/cilium/pkg/k8s/slim/k8s/api/networking/v1"
+)
+
+type serviceAddedEvent struct {
+	service *slim_corev1.Service
+}
+
+type serviceUpdatedEvent struct {
+	oldService *slim_corev1.Service
+	newService *slim_corev1.Service
+}
+
+type serviceDeletedEvent struct {
+	service *slim_corev1.Service
+}
+
+type serviceManager struct {
+	informer cache.Controller
+	store    cache.Store
+
+	queue      workqueue.RateLimitingInterface
+	maxRetries int
+
+	ingressQueue workqueue.RateLimitingInterface
+}
+
+func newServiceManager(ingressQueue workqueue.RateLimitingInterface, maxRetries int) (*serviceManager, error) {
+	manager := &serviceManager{
+		queue:        workqueue.NewRateLimitingQueue(workqueue.DefaultControllerRateLimiter()),
+		ingressQueue: ingressQueue,
+		maxRetries:   maxRetries,
+	}
+
+	manager.store, manager.informer = informer.NewInformer(
+		cache.NewFilteredListWatchFromClient(k8s.WatcherClient().CoreV1().RESTClient(), "services",
+			v1.NamespaceAll, func(options *metav1.ListOptions) {
+				options.LabelSelector = ciliumIngressLabelKey
+			}),
+		&slim_corev1.Service{},
+		0,
+		cache.ResourceEventHandlerFuncs{
+			AddFunc:    manager.handleAddService,
+			UpdateFunc: manager.handleUpdateService,
+			DeleteFunc: manager.handleDeleteService,
+		},
+		nil,
+	)
```

Comment:
@christarazi let's leave this as is because `operator/watchers/service_cache.go` is conditional if kvstore is enabled, `synchronize-k8s-services` is set and this informer has a label selector.

Reply:
Ah thanks, I didn't notice the label selector